src/main/java/nextstep/subway/line/domain/Line.java

```diff
@@ -50,7 +52,73 @@ public String getColor() {
         return color;
     }
 
-    public List<Section> getSections() {
-        return sections;
+    public Sections getSections() {
+        return this.sections;
+    }
+
+    public void addSection(final Section section) {
+        if (!Objects.equals(this, section.getLine())) {
+            section.updateLineBy(this);
+        }
+        if (!sections.isContains(section)) {
+            sections.addSection(section);
+        }
+    }
+
+    public void removeSection(final Station station) {
+        if (this.sections.isSize() <= ONLY_ONE) {
+            throw new IllegalStateException("구간이 한개 뿐이거나 없는 경우에 삭제할수 없습니다.");
+        }
+        removeSectionBy(station);
+    }
+
+    private void removeSectionBy(Station station) {
+        final Optional<Section> upLineStation = sections.getSections().stream().filter(it -> it.isMatchUpStation(station)).findFirst();
```

Comment:
개인적으로는 stream 이후로는 개행을 해주는 것이 가독성에 좋다고 생각하는데
어떻게 생각하시나요? 😄 
```suggestion
        final Optional<Section> upLineStation = sections.getSections().stream()
                .filter(it -> it.isMatchUpStation(station))
                .findFirst();it.isMatchUpStation(station)).findFirst();
```

Reply:
음 stream 이후에는 띄어쓰기 하도록 하겠습니다 ㅎㅎ